openproblems/tasks/spatial_decomposition/methods/rctd.py

```diff
@@ -1,19 +1,52 @@
+from ....tools.conversion import r_function
 from ....tools.decorators import method
-from ....tools.normalize import log_cpm
 from ....tools.utils import check_version
 
-# import rctd, this is an R tool so need to have also R file
+import numpy as np
+import pandas as pd
+
+_rctd = r_function("rctd.R")
 
 
 @method(
-    method_name="Rctd",
+    method_name="RCTD",
     paper_name="Robust decomposition of cell type mixtures in spatial transcriptomics",
     paper_url="https://www.nature.com/articles/s41587-021-00830-w",
     paper_year=2020,
-    code_url="https://github.com/dmcable/RCTD",
+    code_url="https://github.com/almaan/RCTD",
     code_version=check_version("rctd"),
 )
-def stereoscope_log_cpm(adata):
-    log_cpm(adata)
-    # do something
-    return
+def rctd(adata):
+    # exctract single cell reference data
+    sc_adata = adata.uns["sc_reference"].copy()
+    # remove single cell reference from original anndata
+    del adata.uns["sc_reference"]
```

Comment:
are the `del` still necessary?

Reply:
I mainly added it to free up memory, but we can drop it if you thing it's unnecessary -  thinking if we ever add larger single cell reference data sets.